resources/sdk/clisdkclient/extensions/cmd/edges/rebootedge.go

```diff
@@ -24,7 +25,8 @@ var rebootEdgesCmd = &cobra.Command{
 		path := "/api/v2/telephony/providers/edges/{edgeId}/reboot"
 		path = strings.Replace(path, "{edgeId}", args[0], 1)
 
-		results, err := CommandService.Post(path, data)
+		retryFunc := services.RetryWithData(path, data, CommandService.Post)
+		results, err := retryFunc(nil)
```

Comment:
@carnellj The retry configuration goes to default values if it's nil. We can update this to be read from the config file or configured individually in future as required

Reply:
Hi Ronan,

 I liked your implementation exception for one thing.  I would break the RetryFunctions into a separate package.  We are going to handle other Error and Retry handling strategies in the future and I don't want it tied to the services package.  Traditionally, the services package should be for business services code.  This is a support set of functions and it feels these functions could be in their own practice.